### DISCUSSION TITLE
Native Home Assistant MQTT or HACS plugin?

### DISCUSSION THREAD

**Discussion by npawelek** [tagyoureit/nodejs-poolController#933] [Q&A] [answered]
I finally had a chance to get things working this afternoon and I have to say this is awesome! Once getting the controller and configuration in place, I decided to integrate with Home Assistant. I enabled the interface for Home Assistant MQTT and can see during the initial startup, it pushes out the topics but the entities immediately go unavailable. I was looking at the homeassistant.json and noticed that the availability_topic (if I'm reading properly), is probably the culprit because I'm not seeing the `/state/status` topics for the NJS-PC components. Is this working natively for other folks or is it intended that I'll need to modify? I've seen multiple discussions here talk about the Crewski HACS plugin as well, so I'm curious what the general recommendation is? Native MQTT or the alternative?

**Comment by rstrouse** [comment, marked as answer]
You should use the HACS plug-in.  It uses sockets.  https://github.com/Crewski/njsPC-HA
